test/verify/check-storage-luks

```diff
@@ -123,5 +123,80 @@ class TestStorage(StorageCase):
+            self.skipTest("No clevis/tang here")
+
+        # HACK - We generate the keys and cache explicitly before
+        #        starting the socket.  The do get created
+        #        automatically when starting the socket, but
+        #        asynchronously in the background and they might not
+        #        be ready when the socket accepts the first
+        #        connection.
+        #
+        #        https://github.com/latchset/tang/issues/22
+
+        # HACK - systemctl start --wait hangs forever if a condition
+        #        of the unit is not fulfilled.  So we make sure that
+        #        /var/db/tang is empty to avoid surprises.
+
+        m.execute("rm -rf /var/db/tang/*; systemctl start --wait tangd-keygen")
+
+        # HACK - tangd-udpate.service can't create /var/cache/tang.
+        #
+        #        https://github.com/latchset/tang/issues/24
+
+        m.execute("mkdir /var/cache/tang; chown -R tang:tang /var/cache/tang")
+
+        # HACK - /var/cache/tang is automatically updated whenever
+        #        /var/db/tang changes, but also asynchronously in the
+        #        background, and it's not reliable either.
+        #
+        #        https://github.com/latchset/tang/issues/23
+
+        m.execute("systemctl restart tangd-update")
+
+        m.execute("systemctl start tangd.socket")
+        sig_thp = m.execute("curl -s http://localhost/adv | jose fmt -j- -g payload -y -o- | jose jwk use -i- -r -u verify -o- | jose jwk thp -i-").strip()
+
+        self.login_and_go("/storage")
+
+        # Add a disk and format it with luks
+        m.add_disk("50M", serial="MYDISK")
+        b.wait_in_text("#drives", "MYDISK")
+        b.click('tr:contains("MYDISK")')
+        b.wait_visible("#storage-detail")
+
+        self.content_tab_action(1, 1, "Format")
+        self.dialog({ "type": "luks+ext4",
+                      "name": "ENCRYPTED",
+                      "passphrase": "vainu-reku-toma-rolle-kaja",
+                      "passphrase2": "vainu-reku-toma-rolle-kaja" })
+        self.content_row_wait_in_col(1, 1, "Encrypted data")
+        self.content_row_wait_in_col(2, 1, "ext4 File System")
+
+        # Set the flag and reload the page to make the Clevis stuff appear
+        #
+        b.eval_js('window.localStorage["clevis-feature"] = "on"')
+        b.reload()
+        b.enter_page("/storage")
+
+        # Add a key
+        #
+        self.content_tab_wait_in_info(1, 1, "Network keys", "")
+        self.content_tab_action(1, 1, "Add")
+        self.dialog_wait_open()
+        self.dialog_set_val("url", "127.0.0.1")
+        self.dialog_set_val("passphrase", "vainu-reku-toma-rolle-kaja")
+        self.dialog_apply()
+        b.wait_in_text("#dialog", "The output should match this text")
+        b.wait_in_text("#dialog", sig_thp)
+        self.dialog_apply()
+        self.dialog_wait_close()
+        self.content_tab_wait_in_info(1, 1, "Network keys", "127.0.0.1")
```

Comment:
THis doesn't test most of key removal/obsoletion etc. yet, but could you at least lock the device again, unlock it, and make sure this succeeds without typing a passphrase? So that we cover the most basic use case at least.

Reply:
Yes, that would be worthwhile indeed.

Reply:
Now the test cover a lot more, including key rotation.